src/libs/antares/study/parts/hydro/series.cpp

```diff
@@ -193,12 +199,108 @@ void DataSeriesHydro::checkMinGenTsNumber(Study& study, const AreaName& areaID)
     }
 }
 
+bool DataSeriesHydro::LoadHydroPowerCredits(Study& study, const AreaName& areaID, const AnyString& folder)
+{
+    bool ret = true;
+    auto& buffer = study.bufferLoadingTS;
```

Comment:
**bufferLoadingTS** : here we use a buffer held by the study to contain to path of the file we load the data from. And we clear it before  using. What's the point of this if we don't get info from the study ?
We should declare and use the buffer locally, and remove this dependency from the **Study**.

I guess it's the same thing with `study.dataBuffer`, which is a string, and whose responsibility is to hold the text of the file from which we retrieve the data. Then this content is internally interpreted.
Honestly ? This is very low quality code (you're **not** responsible for this, of course, it's done like that in multiple places) : at this code level, we really don't care and don't want to know how the file content is retrieved. This is an internal implement detail that should stay internal. `study.dataBuffer` should not be passed at all !

Reply:
This can be resolved.